cpp/ql/src/semmle/code/cpp/ir/dataflow/DefaultTaintTracking.qll

```diff
@@ -21,6 +21,49 @@ private predicate predictableInstruction(Instruction instr) {
   predictableInstruction(instr.(UnaryInstruction).getUnary())
 }
 
+/**
+ * Functions that we should only allow taint to flow through (to the return
+ * value) if all but the source argument are 'predictable'.  This is done to
+ * emulate the old security library's implementation rather than due to any
+ * strong belief that this is the right approach.
+ *
+ * Note that the list itself is not very principled; it consists of all the
+ * functions listed in the old security library's [default] `isPureFunction`
+ * but not in the old taint tracking library's `returnArgument` predicate.
+ */
+predicate predictableOnlyFlow(string name) {
+  name = "abs" or
+  name = "atof" or
+  name = "atoi" or
+  name = "atol" or
+  name = "atoll" or
+  name = "labs" or
+  name = "strcasestr" or
+  //name = "strcat" or
+  name = "strchnul" or
+  name = "strchr" or
+  name = "strchrnul" or
+  name = "strcmp" or
+  //name = "strcpy" or
+  name = "strcspn" or
+  name = "strdup" or
+  name = "strlen" or
+  //name = "strncat" or
+  name = "strncmp" or
+  //name = "strncpy" or
```

Comment:
I'm a bit confused as a reader about why the string-cat/copy functions are commented out. If it's important to leave them in the list, please add a comment about why.

Reply:
It was intended to make it clearer how the list was constructed, but you're right, I think it just creates confusion (and there's plenty of explanation in the QLDoc now).

Reply:
Done.